ci: auto update generated files

Zxilly opened the pull request:
This can help to make the pkg keep sync with golang as fast as possible.

Zxilly committed:
ci: auto update generated files

Zxilly committed:
chore: update name and cron

Zxilly committed:
chore: update generated files to pass the tests

TcM1911 commented:
The timestamp in the template must be removed if we are to have this workflow. Otherwise, they will be recorded as changed every time this workflow is executed. Can you update the PR?

Zxilly commented:
I think it would be better to set generate to on-demand, I'm working on a patch

Zxilly committed:
feat: update generated files on demand

Zxilly commented:
implemented, plz review

TcM1911 commented on gen.go:
```diff
@@ -154,8 +171,59 @@ type goversion struct {
+	if !diffCode(string(old), string(new)) {
+		fmt.Println(target + " no changes.")
+		return
+	} else {
```
Don't need this else branch since we return in the if branch. Please remove.

TcM1911 commented on gen.go:
```diff
@@ -165,15 +233,17 @@ func processGoVersions() {
 	for *requestURL != "" {
 		fmt.Println("Fetching latests tags")
-		resp, err := client.Get(tagsRequestURL)
+		req, _ := http.NewRequest("GET", *requestURL, nil)
```
Use `http.MethodGet` instead of `"GET"`.

TcM1911 commented on gen.go:
```diff
@@ -217,13 +287,15 @@ func processGoVersions() {
 		}
 
-		resp, err := client.Get(fmt.Sprintf(commitRequestURLFormatStr, tag.Commit.Sha))
+		req, _ := http.NewRequest("GET", fmt.Sprintf(commitRequestURLFormatStr, tag.Commit.Sha), nil)
```
Same here `http.MethodGet`.

TcM1911 commented on gen.go:
```diff
@@ -97,6 +99,21 @@ var goversions = map[string]*GoVersion{
+	token := os.Getenv("GITHUB_TOKEN")
+	if token != "" {
+		authRequest = func(r *http.Request) {
+			r.Header.Set("Authorization", "token "+token)
```
Should be `Bearer` instead of `token`. https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-2-calling-the-rest-api

Zxilly committed:
chore: use guard statement

Zxilly committed:
chore: use http get const

Zxilly committed:
fix: follow github rest api format

https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-2-calling-the-rest-api
